ocaml-eglot.el

```diff
@@ -102,18 +102,71 @@ Otherwise, `merlin-construct' only includes constructors."
 
 ;; Jump to definition
 
-;; TODO: At the moment, the locate is essentially based on
-;; `xref-find-definitions`, which isn't very smart:
-;;
-;; - We don't want to open a new window if the destination is the same
-;;   as the current document.
-;; - We want to control whether we want to jump to ML or MLI
-;; - We'd also like to be able to jump to the definition of a type
-
-(defun ocaml-eglot-locate ()
-  "Locate the identifier at point."
+(defun ocaml-eglot--value-or-type-definition ()
+  "Find the definition (or type definition) to identify it under the cursor."
+  (let ((result (ocaml-eglot-req--definition)))
+    (if (not result)
+        (ocaml-eglot-req--type-definition)
+      result)))
```

Comment:
I don't think we want this fallback. Why would we jump to the definition of the type of an expression when the user asks for the definition of the expression ?

Jump to type declaration should be an its own explicit command `ocaml-eglot-find-type-declaration`

Reply:
Just out of curiosity, why? 
I was very happy to use `C-c C-l` to `locate` in Merlin and I do not understand why I need to have to commands for `definition` and just one for `declaration`?

Reply:
It's three different actions:

1. Jump to the _expression_ definition (`locate` with `ml` target)
1. Jump to the _expression_ declaration (`locate` with `mli` target)
1. Jump to the _type of the expression_ declaration (`locate-type`, actually, there is no bindings for that in the current emacs mode ? Weird.)